test/update_test.go

```diff
@@ -380,6 +380,69 @@ var _ = Describe("update", func() {
 							}))
 						})
 					})
+
+					It("can accept a single file for dependencies", func() {
```

Comment:
this is only testing the "single-file" flow when the `--image` flag is passed in, but i assume it works with or without the flag?

also there's a a lot of duplication with the existing test 🤔 i wonder what's the minimum test that would give us the confidence that it works for a single file... or maybe a table test with the directory case? 🤔

Reply:
The bug was only when `--image` flag was passed in, hence the focus on that test case.

As for the duplication, we unfortunately have different code paths for the two use cases, so I did just copy over the validations. There is maybe space to do a refactoring of the code paths, but they are basically two different if/then cases so I leaned against it.

Let's touch base because happy to go a step further on the refactoring if it makes sense in this scope.